### PR TITLE
Add parallel flag to EQLs

### PR DESCRIPTION
**Pull request by santisoler**
Add the option to enable/disable parallelization on EQLs through a new
`parallel` attribute that can be set on initialization or anytime.
Add dispatcher functions to choose between the regular version or the
parallized version of the `predict_numba` and `jacobian_numba` functions.

**Reminders**:

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst` and the base `__init__.py` file for the package.
- [ ] Write detailed docstrings for all functions/classes/methods. It often helps to design better code if you write the docstrings first.
- [ ] If adding new functionality, add an example to the docstring, gallery, and/or tutorials.
- [ ] Add your full name, affiliation, and [ORCID](https://orcid.org) (optional) to the `AUTHORS.md` file (if you haven't already) in case you'd like to be listed as an author on the [Zenodo](https://zenodo.org/communities/fatiando) archive of the next release.

### CODE REVIEW

**Comment by santisoler** [comment]
Todo:
- [ ] Add `parallel` flag to `EQLHarmonicSpherical`
- [ ] Write test functions for checking if the the processes are running in parallel or not. How to do this?

File: harmonica/equivalent_layer/utils.py [hunk 1]
```diff
@@ -59,6 +59,37 @@ def predict_numba(
...
+def dispatch_jacobian(parallel):
+    """
+    Return either the parallel or non-parallel jacobian function
+    """
+    if parallel:
+        return jacobian_numba_parallel
+    else:
+        return jacobian_numba
+
+
```
Comment (leouieda) [inline harmonica/equivalent_layer/utils.py:92]: This coule be replaced with a dictionary inside the class:
 
```python

class EQLHarmonic(...):
    self._predict_kernel = {False: _predict_numba_serial, False: _predict_numba_parallel}

    def predict(...):
        self._predict_kernel[self.parallel](coordinates, ...)
```

File: harmonica/equivalent_layer/utils.py [hunk 2]
```diff
@@ -59,6 +59,37 @@ def predict_numba(
...
 
+# Define parallelized version of the utils functions
+if hasattr(predict_numba, "py_func"):
+    predict_numba_parallel = jit(nopython=True, parallel=True)(predict_numba.py_func)
+else:
+    predict_numba_parallel = jit(nopython=True, parallel=True)(predict_numba)
+if hasattr(jacobian_numba, "py_func"):
+    jacobian_numba_parallel = jit(nopython=True, parallel=True)(jacobian_numba.py_func)
+else:
+    jacobian_numba_parallel = jit(nopython=True, parallel=True)(jacobian_numba)
```
Comment (leouieda) [inline harmonica/equivalent_layer/utils.py:70]: What about not putting the decorators on the functions and doing:

```python
predict_numba = numba.jit(nopython=True)(predict)
predict_numba_parallel = numba.jit(nopython=True, parallel=True)(predict)
```

This way, there is not `if` block or the cryptic inspection of attributes (which numba might decide to change).

Comment (santisoler) [reply harmonica/equivalent_layer/utils.py:70]: You're right about this. The whole `if hasattr(...)` workaround was intended to solve the design error of keeping the `jit` decorator on the functions. Not my best week haha!

File: harmonica/equivalent_layer/utils.py [hunk 3]
```diff
@@ -59,10 +70,8 @@ def predict_numba(
...
 
-def pop_extra_coords(kwargs):
-    """
-    Remove extra_coords from kwargs
-    """
-    if "extra_coords" in kwargs:
-        warn("EQL gridder will ignore extra_coords: {}.".format(kwargs["extra_coords"]))
-        kwargs.pop("extra_coords")
+# pylint: disable=invalid-name
+predict_numba_serial = numba.jit(nopython=True)(predict)
```
Comment (santisoler) [inline harmonica/equivalent_layer/utils.py:74]:
```suggestion
predict_numba_serial = numba.jit(nopython=False)(predict)
```

**Pull request merged**